ld: library not found for -lgfortran MacOS, clang + ifort compilers

f_check has

```
if ($vendor eq "INTEL"){
    $linker_a .= "-lgfortran"
}
```

I don't even have GNU compilers on my machine. 

If I remove these lines then I get 

```
mpif90 -O2 -fPIC  -all_load -headerpad_max_install_names -install_name "/Users/barrysmith/Src/petsc/arch-maint/externalpackages/git.openblas/exports/../libopenblas.0.dylib" -dynamiclib -o ../libopenblas_haswellp-r0.3.8.dev.dylib ../libopenblas_haswellp-r0.3.8.dev.a -Wl,-exported_symbols_list,osx.def  -L/Users/barrysmith/soft/clang-ifort/lib -L/Users/barrysmith/soft/clang-ifort/lib -Wl,-rpath,/opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/tbb/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/mkl/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/tbb/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/mkl/lib -L/opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib -L/usr/lib  -lmpifort -lmpi -lpmpi -lmpifort -lmpi -lpmpi -lSystem -lpthread  /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libifport.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libifcoremt.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libimf.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libsvml.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libipgo.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libirc.a /opt/intel/compilers_and_libraries_2019.4.233/mac/compiler/lib/libsvml.a /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/lib/darwin/libclang_rt.osx.a 
make[1]: Leaving directory '/Users/barrysmith/Src/petsc/arch-maint/externalpackages/git.openblas/exports'Undefined symbols for architecture x86_64:
  "_MAIN__", referenced from:
      _main in libifcoremt.a(for_main.o)
  "__FTN_ALLOC", referenced from:
      _do_ftn_alloc in libifcoremt.a(for_ftnalloc.o)
ld: symbol(s) not found for architecture x86_64
make[1]: *** [Makefile:123: libopenblas_haswellp-r0.3.8.dev.dylib] Error 1
```

The problem is that you are not stripping out the libifcoremt.a library so the resulting list cannot be used to make a dynamic library

Mac OS 10.14.6 
ifort version 19.0.4.233
Apple clang version 11.0.0 (clang-1100.0.33.12)

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -4,7 +4,15 @@ trigger:
   branches:
     include:
       - develop
-
+resources:
+  containers:
+      - container: oneapi-hpckit
+        image: intel/oneapi-hpckit:latest
+        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
+      - container: oneapi-basekit
+        image: intel/oneapi-basekit:latest
+        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
+ 
 jobs:
 # manylinux1 is useful to test because the
 # standard Docker container uses an old version
@@ -97,7 +105,7 @@ jobs:
       brew tap LouisBrunner/valgrind
       brew install --HEAD LouisBrunner/valgrind/valgrind
       make TARGET=SANDYBRIDGE NO_AVX512=1 USE_OPENMP=1 INTERFACE64=1 DYNAMIC_ARCH=1 DYNAMIC_LIST=SANDYBRIDGE DEBUG=1 NO_PARALLEL_MAKE=1 CC=/usr/local/opt/llvm/bin/clang FC=gfortran-10
- 
+
 - job: OSX_Ifort_Clang
   pool:
      vmImage: 'macOS-10.15'
@@ -106,15 +114,7 @@ jobs:
      MACOS_HPCKIT_URL: https://registrationcenter-download.intel.com/akdlm/irc_nas/17643/m_HPCKit_p_2021.2.0.2903_offline.dmg
      LIBRARY_PATH: /usr/local/opt/llvm/lib
      MACOS_FORTRAN_COMPONENTS: intel.oneapi.mac.ifort-compiler
-  resources:
-    containers:
-      - container: oneapi-hpckit
-        image: intel/oneapi-hpckit:latest
-        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
-      - container: oneapi-basekit
-        image: intel/oneapi-basekit:latest
-        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
-  steps:   
+ steps:   
   - script: |
       brew update
       brew install llvm libomp

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -136,5 +136,6 @@ jobs:
     displayName: install
     condition: ne(variables.CACHE_RESTORED, 'true')
   - script: | 
+      source /opt/intel/oneapi/setvars.sh
       make CC=/usr/local/opt/llvm/bin/clang FC=ifort
  

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -133,5 +133,6 @@ jobs:
       hdiutil detach /Volumes/"$(basename "$URL" .dmg)" -quiet
     displayName: install
     condition: ne(variables.CACHE_RESTORED, 'true')
+  - script: | 
       make CC=/usr/local/opt/llvm/bin/clang FC=ifort
  

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -97,4 +97,42 @@ jobs:
       brew tap LouisBrunner/valgrind
       brew install --HEAD LouisBrunner/valgrind/valgrind
       make TARGET=SANDYBRIDGE NO_AVX512=1 USE_OPENMP=1 INTERFACE64=1 DYNAMIC_ARCH=1 DYNAMIC_LIST=SANDYBRIDGE DEBUG=1 NO_PARALLEL_MAKE=1 CC=/usr/local/opt/llvm/bin/clang FC=gfortran-10
-      cd ctest; OMP_NUM_THREADS=1 valgrind ./xscblat2 <sin2
+ 
+- job: OSX_Ifort_Clang
+  pool:
+     vmImage: 'macOS-10.15'
+  variables:
+     LD_LIBRARY_PATH: /usr/local/opt/llvm/lib
+     MACOS_HPCKIT_URL: https://registrationcenter-download.intel.com/akdlm/irc_nas/17643/m_HPCKit_p_2021.2.0.2903_offline.dmg
+     LIBRARY_PATH: /usr/local/opt/llvm/lib
+     MACOS_FORTRAN_COMPONENTS: intel.oneapi.mac.ifort-compiler
+  resources:
+    containers:
+      - container: oneapi-hpckit
+        image: intel/oneapi-hpckit:latest
+        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
+      - container: oneapi-basekit
+        image: intel/oneapi-basekit:latest
+        options: '-v /usr/bin/sudo:/usr/bin/sudo -v /usr/lib/sudo/libsudo_util.so.0:/usr/lib/sudo/libsudo_util.so.0 -v /usr/lib/sudo/sudoers.so:/usr/lib/sudo/sudoers.so'
+  steps:   
+  - script: |
+      brew update
+      brew install llvm libomp
+  - script: |
+      sudo mkdir -p /opt/intel
+      sudo chown $USER /opt/intel
+    displayName: prepare for cache restore
+  - task: Cache@2
+    inputs:
+      path: /opt/intel/oneapi
+      key: '"install" | "$(MACOS_HPCKIT_URL)" | "$(MACOS_FORTRAN_COMPONENTS)"'
+      cacheHitVar: CACHE_RESTORED
+  - script: 
+      curl --output webimage.dmg --url $(MACOS_HPCKIT_URL) --retry 5 --retry-delay 5
+      hdiutil attach webimage.dmg
+      sudo /Volumes/"$(basename "$(MACOS_HPCKIT_URL)" .dmg)"/bootstrapper.app/Contents/MacOS/bootstrapper -s --action install --components="$(MACOS_FORTRAN_COMPONENTS)" --eula=accept --continue-with-optional-error=yes --log-dir=.
+      hdiutil detach /Volumes/"$(basename "$URL" .dmg)" -quiet
+    displayName: install
+    condition: ne(variables.CACHE_RESTORED, 'true')
+      make CC=/usr/local/opt/llvm/bin/clang FC=ifort
+ 

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -127,7 +127,7 @@ jobs:
       path: /opt/intel/oneapi
       key: '"install" | "$(MACOS_HPCKIT_URL)" | "$(MACOS_FORTRAN_COMPONENTS)"'
       cacheHitVar: CACHE_RESTORED
-  - script: 
+  - script: | 
       curl --output webimage.dmg --url $(MACOS_HPCKIT_URL) --retry 5 --retry-delay 5
       hdiutil attach webimage.dmg
       sudo /Volumes/"$(basename "$(MACOS_HPCKIT_URL)" .dmg)"/bootstrapper.app/Contents/MacOS/bootstrapper -s --action install --components="$(MACOS_FORTRAN_COMPONENTS)" --eula=accept --continue-with-optional-error=yes --log-dir=.

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -102,9 +102,7 @@ jobs:
   - script: |
       brew update
       brew install llvm libomp
-      brew tap LouisBrunner/valgrind
-      brew install --HEAD LouisBrunner/valgrind/valgrind
-      make TARGET=SANDYBRIDGE NO_AVX512=1 USE_OPENMP=1 INTERFACE64=1 DYNAMIC_ARCH=1 DYNAMIC_LIST=SANDYBRIDGE DEBUG=1 NO_PARALLEL_MAKE=1 CC=/usr/local/opt/llvm/bin/clang FC=gfortran-10
+      make TARGET=CORE2 USE_OPENMP=1 INTERFACE64=1 DYNAMIC_ARCH=1 CC=/usr/local/opt/llvm/bin/clang FC=gfortran-10
 
 - job: OSX_Ifort_Clang
   pool:

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -118,7 +118,6 @@ jobs:
   - script: |
       brew update
       brew install llvm libomp
-  - script: |
       sudo mkdir -p /opt/intel
       sudo chown $USER /opt/intel
     displayName: prepare for cache restore

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -114,7 +114,7 @@ jobs:
      MACOS_HPCKIT_URL: https://registrationcenter-download.intel.com/akdlm/irc_nas/17643/m_HPCKit_p_2021.2.0.2903_offline.dmg
      LIBRARY_PATH: /usr/local/opt/llvm/lib
      MACOS_FORTRAN_COMPONENTS: intel.oneapi.mac.ifort-compiler
- steps:   
+  steps:   
   - script: |
       brew update
       brew install llvm libomp

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -130,7 +130,9 @@ jobs:
       curl --output webimage.dmg --url $(MACOS_HPCKIT_URL) --retry 5 --retry-delay 5
       hdiutil attach webimage.dmg
       sudo /Volumes/"$(basename "$(MACOS_HPCKIT_URL)" .dmg)"/bootstrapper.app/Contents/MacOS/bootstrapper -s --action install --components="$(MACOS_FORTRAN_COMPONENTS)" --eula=accept --continue-with-optional-error=yes --log-dir=.
+      installer_exit_code=$?
       hdiutil detach /Volumes/"$(basename "$URL" .dmg)" -quiet
+      exit $installer_exit_code
     displayName: install
     condition: ne(variables.CACHE_RESTORED, 'true')
   - script: | 

--- a/exports/Makefile
+++ b/exports/Makefile
@@ -140,7 +140,11 @@ ifneq (,$(filter 1 2,$(NOFORTRAN)))
 #only build without Fortran
 	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
 else
+ifeq ($F_COMPILER, INTEL)
+	$(FC) $(FFLAGS) $(LDFLAGS) -all-load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
+else
 	$(FC) $(FFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
+endif
 endif
 
 dllinit.$(SUFFIX) : dllinit.c

--- a/exports/Makefile
+++ b/exports/Makefile
@@ -140,7 +140,7 @@ ifneq (,$(filter 1 2,$(NOFORTRAN)))
 #only build without Fortran
 	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
 else
-ifeq ($F_COMPILER, INTEL)
+ifeq ($(F_COMPILER), INTEL)
 	$(FC) $(FFLAGS) $(LDFLAGS) -all-load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
 else
 	$(FC) $(FFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)

--- a/f_check
+++ b/f_check
@@ -391,10 +391,6 @@ if ($link ne "") {
 
 }
 
-if ($vendor eq "INTEL"){
-    $linker_a .= "-lgfortran"
-}
-
 if ($vendor eq "FLANG"){
     $linker_a .= "-lflang"
 }

